src/main.js

```diff
@@ -511,6 +519,7 @@ var manifestor = function(options) {
   function selectCanvas(item) {
     var state = canvasState();
     state.selectedCanvas = item;
```

Comment:
The mode switching checks for the previous state in order to determine the "direction" of the transition ("from" detail "to" overview, etc. This is important to set on all modal changes in order to distinguish between the first and subsequent layouts. The initial layout will have no previous state.